lib/protobuf/field/field_hash.rb

```diff
@@ -0,0 +1,97 @@
+module Protobuf
+  module Field
+    class FieldHash < Hash
+
+      ##
+      # Attributes
+      #
+
+      attr_reader :field, :key_field, :value_field
+
+      ##
+      # Constructor
+      #
+
+      def initialize(field)
+        @field = field
+        @key_field = field.type_class.get_field(:key)
+        @value_field = field.type_class.get_field(:value)
+      end
+
+      ##
+      # Public Instance Methods
+      #
+
+      def []=(key, val)
+        super(normalize_key(key), normalize_val(val)) unless val.nil?
+      end
+
+      def store(key, val)
+        super(normalize_key(key), normalize_val(val)) unless val.nil?
+      end
+
+      def replace(val)
+        raise_type_error(val) unless val.is_a?(Hash)
+        clear
+        update(val)
+      end
+
+      def merge!(other)
+        raise_type_error(other) unless other.is_a?(Hash)
+        other.each { |k, v| self[normalize_key(k)] = normalize_val(v) }
+      end
+
+      def update(other)
```

Comment:
`merge!` and `update` look the same. Again, just alias the method name (that or leave only one of them)?